### PR TITLE
Fix name signature bug and extend test for instance fields

### LINKED ISSUE #2923: Add tests for PR #2916
* Add tests for #2916 
* See my comment on the PR page

### PR DESCRIPTION
**Pull request by janmasrovira**
- Closes #2923 

This pr fixes a bug where all fields were assigned to be explicit arguments in the NameSignature Builder. A single line change was enough to fix it.
```diff
-           RecordStatementField RecordField {..} -> addSymbol @s Explicit Nothing _fieldName _fieldType
+           RecordStatementField RecordField {..} -> addSymbol @s (fromIsImplicitField _fieldIsImplicit) Nothing _fieldName _fieldType
```

I've also added a compilation test for instance fields.

### CODE REVIEW

File: test/Compilation/Positive.hs [hunk 1]
```diff
@@ -455,5 +455,10 @@ tests =
         "Test076: Builtin Maybe"
         $(mkRelDir ".")
         $(mkRelFile "test076.juvix")
-        $(mkRelFile "out/test076.out")
+        $(mkRelFile "out/test076.out"),
+      posTestEval
```
Comment (paulcadman) [inline test/Compilation/Positive.hs:459]: Can this be `posTest` instead of `posTestEval` so we test compilation post-core?

Comment (janmasrovira) [reply test/Compilation/Positive.hs:459]: my understanding is that after core it doesn't make a difference wether a field is an instance field or not. Should I still change it to `posTest`?

Comment (paulcadman) [reply test/Compilation/Positive.hs:459]: We can keep it like this if you like. The reason to change it is to add more test coverage to the post-core pipeline.

**Review by paulcadman — APPROVED** [review]

File: test/Compilation/Positive.hs (hunk 1 above)
Comment (janmasrovira) [reply test/Compilation/Positive.hs]: I've made the change. I think it is an overall relevant test.

**Review by paulcadman — APPROVED** [review]

**Pull request merged**